src/extension/importer/core/src/main/java/org/geoserver/importer/SpatialFile.java

```diff
@@ -99,36 +101,42 @@ public String apply(@Nullable StyleHandler input) {
 
         // getBaseName only gets the LAST extension so beware for .shp.aux.xml stuff
         final String baseName = getBaseName(file.getName());
+        final String baseExtension = getExtension(file.getName());
+        final String basePath = file.getParent();
+
+        // look for style files
```

Comment:
I would add a comment explaining why it's done this way, on a folder with few files this approach is actually more work. However, it's not that much more work, while on a directory with lots of files, this is more efficient.

Reply:
Done